.github/actions/query_issues/action.yml

```
@@ -0,0 +1,40 @@
+name: Query issues using GraphQL API
+inputs:
+  token:
+    description: 'The GITHUB_TOKEN secret'
+    required: true
+  query:
+    description: "The GraphQL search query to be executed"
```

The name/description here are not very helpful. Initially I thought you were passing in some nested GraphQL query. But apparently these are just the search terms you can use in issue search box on Github?

How about this?
```suggestion
  filter:
    # Filter syntax: https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests
    description: "The search filter for issues and/or PRs."
```

---

Actually, given that you use `on Issue` below, can you filter for PRs here or will it only match issues?

---

It will only match issues. But not only because of the filter but also because of the query:
```
                 nodes {
                    ... on Issue {
                      title
                      url
                    }
                  }
```
We can change the action to be more generic and accept a graphql query as input instead of the filter.
Not sure if it is worth though. The advantage is that we could use it in other repositories as well.